daemon/images/image_pull.go

```diff
@@ -50,6 +52,18 @@ func (i *ImageService) PullImage(ctx context.Context, image, tag string, platfor
 
 	err = i.pullImageWithReference(ctx, ref, platform, metaHeaders, authConfig, outStream)
 	imageActions.WithValues("pull").UpdateSince(start)
+
+	if err == nil && platform != nil {
+		// --platform was specified: print warning for single-arch images that do not match
+		img, err := i.GetImage(image, platform)
+		if errdefs.IsNotFound(err) && img != nil {
+			po := streamformatter.NewJSONProgressOutput(outStream, false)
+			progress.Messagef(po, "", `WARNING: %s`, err.Error())
+			logrus.WithError(err).WithField("image", image).Warn("ignoring platform mismatch on single-arch image")
+			err = nil
```

Comment:
I'm not sure I understand this line.

Reply:
LOL, I literally took that from your PR https://github.com/moby/moby/pull/41873 😅 

But to answer your question; `err` is not "local" to this `if`, so we need to reset it as we only want a warning to be printed, and not make the pull fail.

Alternatively I could use a local err (e.g. `iErr`) if you think that's better

Reply:
A local `iErr` makes more sense to me as it's harder to miss when changing the code in the future.

Reply:
I updated this code, and now use an early return, as well as added more comments to describe what we're doing